pkg/certificates/certs.go

```diff
@@ -142,7 +127,7 @@ func CreateCACerts(ctx context.Context, expirationInterval time.Duration) (*KeyP
 }
 
 // CreateControlPlaneCert generates the certificate for the client
-func CreateControlPlaneCert(ctx context.Context, caKey *rsa.PrivateKey, caCertificate *x509.Certificate, expirationInterval time.Duration) (*KeyPair, error) {
+func CreateControlPlaneCert(ctx context.Context, caKey *rsa.PrivateKey, caCertificate *x509.Certificate, expirationInterval time.Duration, san string) (*KeyPair, error) {
```

Comment:
I'd suggest not making a breaking change in the library.  Two options:

1. Make this `san ...string`, and have empty string maintain the current behavior.
2. Add a _new_ function, e.g. `CreateServerCert` which takes the new arguments.

When we make breaking changes in these libraries, it gums up the dependency / buoy process for upgrading all the clients.

We could still remove this function (if we want) after we've migrated the callers, but I'd prefer not to get in the habit of making breaking library changes in broad dependencies (and, sadly, `control-protocol` has broad dependencies).

Reply:
nice. 
Let's g with option 1